README.md

```diff
@@ -539,6 +539,136 @@ val jsonSchema: ASchema = TapirSchemaToJsonSchema(
 )
 ```
 
+#### Generating JSON Schema from case class
+
+We can also generate JSON Schema directly from case class, without defining the schema manually:
+
+```scala mdoc:compile-only
+//> using dep com.softwaremill.sttp.openai::core:0.3.1
+
+import sttp.openai.OpenAISyncClient
+import sttp.openai.json.SnakePickle
+import sttp.openai.requests.completions.chat.ChatRequestBody.ChatBody
+import sttp.openai.requests.completions.chat.ChatRequestBody.ChatCompletionModel.GPT4oMini
+import sttp.openai.requests.completions.chat.ToolCall.FunctionToolCall
+import sttp.openai.requests.completions.chat.message.Content.TextContent
+import sttp.openai.requests.completions.chat.message.Message.{AssistantMessage, ToolMessage, UserMessage}
+import sttp.openai.requests.completions.chat.message.Tool.SchematizedFunctionTool
+import sttp.tapir.generic.auto._
+
+case class Passenger(name: String, age: Int)
+
+object Passenger {
+  implicit val passengerR: SnakePickle.Reader[Passenger] = SnakePickle.macroR[Passenger]
+}
+
+case class FlightDetails(passenger: Passenger, departureCity: String, destinationCity: String)
+
+object FlightDetails {
+  implicit val flightDetailsR: SnakePickle.Reader[FlightDetails] = SnakePickle.macroR[FlightDetails]
+}
+
+case class BookedFlight(confirmationNumber: String, status: String)
+
+object BookedFlight {
+  implicit val bookedFlightW: SnakePickle.Writer[BookedFlight] = SnakePickle.macroW[BookedFlight]
+}
+
+object Main extends App {
+  val apiKey = System.getenv("OPENAI_KEY")
+  val openAI = OpenAISyncClient(apiKey)
+
+  val initialRequestMessage = Seq(UserMessage(content = TextContent("I want to book a flight from London to Tokyo for Jane Doe, age 34")))
+
+  val givenRequest = ChatBody(
```

Comment:
let's add some comments here describing how this differs from a "normal" open ai request, and what amount of labor we are saving